libbeat/cmd/root.go

```diff
@@ -52,22 +54,48 @@ type BeatsRootCmd struct {
 	KeystoreCmd   *cobra.Command
 }
 
-// GenRootCmd returns the root command to use for your beat. It takes
-// beat name as parameter, and also run command, which will be called if no args are
-// given (for backwards compatibility)
+// GenRootCmd returns the root command to use for your beat. It takes the beat name, version,
+// and run command, which will be called if no args are given (for backwards compatibility)
+// DEPRECATED! Use GenRootCmdWithSettings instead.
 func GenRootCmd(name, version string, beatCreator beat.Creator) *BeatsRootCmd {
 	return GenRootCmdWithRunFlags(name, version, beatCreator, nil)
 }
 
 // GenRootCmdWithRunFlags returns the root command to use for your beat. It takes
-// beat name as parameter, and also run command, which will be called if no args are
-// given (for backwards compatibility). runFlags parameter must the flagset used by
+// beat name, version, run command, and runFlags. runFlags parameter must the flagset used by
 // run command
+// DEPRECATED! Use GenRootCmdWithSettings instead.
 func GenRootCmdWithRunFlags(name, version string, beatCreator beat.Creator, runFlags *pflag.FlagSet) *BeatsRootCmd {
 	return GenRootCmdWithIndexPrefixWithRunFlags(name, name, version, beatCreator, runFlags)
 }
 
+// GenRootCmdWithIndexPrefixWithRunFlags returns the root command to use for your beat. It takes
+// beat name, index prefix, version, run command, and runFlags. runFlags parameter must the flagset used by
+// run command
+// DEPRECATED! Use GenRootCmdWithSettings instead.
```

Comment:
This is no valid note marker. See: https://blog.golang.org/godoc-documenting-go-code

It's some badly documented convention, but godoc can parse and filter out/display notes. The syntax for notes is: `// Marker(id): description`.



In this case the comment should say:

```
// GenRootCmdWithIndexPrefixWithRunFlags returns the root command to use for your beat. It takes
// beat name, index prefix, version, run command, and runFlags. runFlags parameter must the flagset used by
// run command
//
// Deprecated: Use GenRootCmdWithSettings instead.
```

For conventions on `Deprecation: ` use see: https://github.com/golang/go/issues/10909#issuecomment-136492606